### PR TITLE
refactor collectImportedServices func

### PR DESCRIPTION
**Pull request by wulianglongrd**
This PR doesn't have any optimizations other than improving readability, but it makes sense to me. If you have any suggestions, please feel free to leave a comment. Thank you for your review.

### CODE REVIEW

File: pilot/pkg/model/sidecar.go [hunk 1]
```diff
@@ -313,117 +313,107 @@ func convertToSidecarScope(ps *PushContext, sidecarConfig *config.Config, config
...
-	for _, listener := range out.EgressListeners {
+	for _, listener := range sc.EgressListeners {
 		// First add the explicitly requested services, which take priority
 		for _, s := range listener.services {
-			out.appendSidecarServices(servicesAdded, s)
+			sc.appendSidecarServices(servicesAdded, s)
 		}
 		// add dependencies on delegate virtual services
-		delegates := ps.DelegateVirtualServices(listener.virtualServices)
-		for _, delegate := range delegates {
```
Comment (wulianglongrd) [inline pilot/pkg/model/sidecar.go:324]: We don't need a for loop here.

File: pilot/pkg/model/sidecar.go [hunk 2]
```diff
@@ -313,117 +313,107 @@ func convertToSidecarScope(ps *PushContext, sidecarConfig *config.Config, config
...
 		// want in the hosts field, and the potentially random choice below won't matter
 		for _, vs := range listener.virtualServices {
 			for _, cfg := range VirtualServiceDependencies(vs) {
-				out.AddConfigDependencies(cfg.HashCode())
+				sc.AddConfigDependencies(cfg.HashCode())
 			}
 
 			v := vs.Spec.(*networking.VirtualService)
 			for h, ports := range virtualServiceDestinations(v) {
+				byNamespace := ps.ServiceIndex.HostnameAndNamespace[host.Name(h)]
```
Comment (wulianglongrd) [inline pilot/pkg/model/sidecar.go:357]: shared `byNamespace` variable.

**Comment by wulianglongrd** [comment]
To be honest, I'm not sure if this PR will make sense to other people, but it makes it more comfortable for me to read. If not, I'll happily close it.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**